framework/base/Controller.php

```diff
@@ -270,8 +270,8 @@ public function createAction($id)
      */
     public function beforeAction($action)
     {
-        $event = new ActionEvent($action);
-        $this->trigger(self::EVENT_BEFORE_ACTION, $event);
+        $event = new ActionEvent($action, ['name' => self::EVENT_BEFORE_ACTION]);
```

Comment:
What do you think about using classes for events same as we do for exceptions:

```php
class BeforeActionEvent extends Event
{
    const NAME = 'request.action.before';

    public function getName()
    {
        return self::NAME;
    }
}
```

It would look like:

```php
$event = new BeforeActionEvent($action);
```

Reply:
Mostly such classes are redundant.
We already have specific event classes for some cases like `ActionEvent` or `ModelEvent`.
I think introduction of the separated class just because it provides default event name is a redundant overcomplication.

Reply:
OK.